src/google/protobuf/compiler/cpp/cpp_bootstrap_unittest.cc

```diff
@@ -183,6 +183,69 @@ TEST(BootstrapTest, GeneratedFilesMatch) {
   }
 }
 
+//test Generate in cpp_generator.cc
+TEST(BootstrapTest, optionNotExist) {
+  CodeGenerator *generator=new cpp::CppGenerator();
+  DescriptorPool pool_;
+  GeneratorContext *generator_context;
+  const std::string &parameter_="aaa";
+  string error;
+  ASSERT_FALSE(generator->Generate(pool_.FindFileByName("google/protobuf/descriptor.proto"),
+                                   parameter_, generator_context, &error));
+  EXPECT_EQ(error,"Unknown generator option: "+parameter_);
+}
+
+//test Generate in cpp_generator.cc
+TEST(BootstrapTest, optionIsNull) {
+  CodeGenerator *generator=new cpp::CppGenerator();
+  DescriptorPool pool_;
+  GeneratorContext *generator_context;
+  const std::string &parameter_=" ";
+  string error;
+  // when parameter_=null,here will return false
+  ASSERT_FALSE(generator->Generate(pool_.FindFileByName("google/protobuf/descriptor.proto"),
+                                   parameter_, generator_context, &error));
+  EXPECT_EQ(error,"Unknown generator option: "+parameter_);
+}
+
+//test Generate in cpp_generator.cc
+TEST(BootstrapTest, optionIsSafeBoundaryChecksafeBoundaryCheck) {
+  CodeGenerator *generator=new cpp::CppGenerator();
+  DescriptorPool pool_;
+  GeneratorContext *generator_context;
+  const std::string &parameter_="safe_boundary_check";
+  string error;
+  ASSERT_FALSE(generator->Generate(pool_.FindFileByName("google/protobuf/descriptor.proto"), 
+                                   parameter_, generator_context, &error));
+  EXPECT_EQ(error,"The safe_boundary_check option is not supported outside of Google.");
+}
+
+//test Generate in cpp_generator.cc
+TEST(BootstrapTest, optionReturnTrue) {
```

Comment:
@acozzette  How about this case?  Maybe it is not duplicated with others

Reply:
I think the options are already fairly well covered since we rely on them all the time and would notice pretty quickly if they stopped working. Also I hesitate to expand the scope of this test file too much, since its main purpose is just to check that the generated files are up to date.

Reply:
> I think the options are already fairly well covered since we rely on them all the time and would notice pretty quickly if they stopped working. Also I hesitate to expand the scope of this test file too much, since its main purpose is just to check that the generated files are up to date.

OK, thank you for pointing out, I understand your concern. 
The commit has already updated.